### PR TITLE
Pass wait group on the stop channel, to indicate when stopped

### PR DESCRIPTION
**Pull request by neiljerram**

### CODE REVIEW

**Comment by caseydavenport** [comment]
@neiljerram is there a bug that this fixes? 

Could you update the `docs` and `release-note` labels accordingly? I'm not sure what the impact of this PR is, otherwise I'd do it myself :)

**Comment by neiljerram** [comment]
@caseydavenport This is to do with Felix controlled shutdown, where we now want to be sure that Felix writes a file containing learnt DNS information, before it exits. Previously the only case of Felix doing something before shutdown was calling the Binder, provided by this repo, to close down sockets used for communication with Dikastes running in local pods. But the pattern didn't ensure that any shutdown actions actually completed before Felix then exited; the point of the wait group in this change is to ensure that.

Now, for this repo, and the Binder, it probably doesn't matter much, because I think the OS would close sockets on Felix exit anyway. But a pattern like this is needed for the DNS info file, and I changed this repo too so as to have a consistent interface between the Felix top level and its various subcomponents.

Do we need to release note this repo specifically? Does pod2daemon have a wider purpose than just being part of Felix and ALP?

**Comment by caseydavenport** [comment]
> Do we need to release note this repo specifically? Does pod2daemon have a wider purpose than just being part of Felix and ALP?

The release notes are actually pulled into the main Calico release, not per-repository, so I don't think we need to release note this repo specifically.

> Now, for this repo, and the Binder, it probably doesn't matter much, because I think the OS would close sockets on Felix exit anyway

OK cool, that's what I would have thought. In that case, I don't think this PR has user-facing impacts so probably doesn't need docs or a release note.

File: binder/binder.go [hunk 1]
```diff
@@ -38,7 +39,7 @@ type Binder interface {
 
 	// Search for pod mounts to bind sockets in.
 	// Send any value over the stop channel to gracefully cancel.
-	SearchAndBind(stop <-chan bool)
+	SearchAndBind(stop <-chan *sync.WaitGroup)
```
[the remark this answers is not in the document]
Comment (neiljerram) [reply binder/binder.go]: Thanks Spike, will do.

**Pull request merged**